### ISSUE TITLE
Configuration file when settings are changed in GUI

### ISSUE THREAD

**Issue by Pantamis** [bitcoin/bitcoin#17692]
Hi !

This issue is related to settings that seems to be different when running `bitcoind` and `bitcoin-qt` on my laptop.

Thanks for help !

**Is your feature request related to a problem? Please describe.**

When settings are changed in `bitcoin-qt`, no configuration file is created. This has several consequences : 
   - `bitcoind` cannot work properly anymore, particulary in the case of a pruned node where it shutdown at start returning `You need to rebuild the database using -reindex to go back to unpruned mode. This will redownload the entire blockchain.
Please restart with -reindex or -reindex-chainstate to recover.`. `bitcoin-qt` will start normally because it knows that the node is pruned, I have no idea how since it did not write it in  any `bitcoin.conf` (there must be a configuration file somewhere).
   - It is hard to review exactly with what settings the node is running if `bitcoin-qt` has been ran once. In my case, I suspect I have difficulties to run Electrum-Personal-Server (EPS) in multiwallet mode because of some hidden configuration settings (related to the issue https://github.com/chris-belcher/electrum-personal-server/issues/111)

**Describe the solution you'd like**

Ideally, all settings changes made in GUI should be reported directly in configuration file `bitcoin.conf` but `bitcoin-qt` could have it's own `bitcoin-qt.conf`. We just need to make it clear about what are the settings used by qt.

**Describe alternatives you've considered**

Settings used for `bitcoin-qt` must be stored somewhere. I did not find any related configuration file other than `bitcoin.conf` in datadir and `.bitcoin` folder. It would be enough to add the location of such file in documentation. 

I am on Ubuntu 18.04, I changed the default datadir, it is well detected by `bitcoind` and `bitcoin-qt` (and the GUI is opening my own config file correctly when I asked him to open the file). But `bitcoind` is not working while `bitcoin-qt` does.

**Additional context**

Here the logs for `bitcoind` where it shutdowns because of pruning : 

2019-12-07T15:32:36Z Bitcoin Core version v0.19.0.1 (release build)
2019-12-07T15:32:36Z Assuming ancestors of block 00000000000000000005f8920febd3925f8272a6a71237563d78c2edfdd09ddf have valid signatures.
2019-12-07T15:32:36Z Setting nMinimumChainWork=000000000000000000000000000000000000000008ea3cf107ae0dec57f03fe8
2019-12-07T15:32:36Z Using the 'sse4(1way),sse41(4way),avx2(8way)' SHA256 implementation
2019-12-07T15:32:36Z Using RdRand as an additional entropy source
2019-12-07T15:32:36Z Default data directory /home/login/.bitcoin
2019-12-07T15:32:36Z Using data directory /home/login/Bitcoin
2019-12-07T15:32:36Z Config file: /home/login/Bitcoin/bitcoin.conf
2019-12-07T15:32:36Z Using at most 125 automatic connections (1024 file descriptors available)
2019-12-07T15:32:36Z Using 16 MiB out of 32/2 requested for signature cache, able to store 524288 elements
2019-12-07T15:32:36Z Using 16 MiB out of 32/2 requested for script execution cache, able to store 524288 elements
2019-12-07T15:32:36Z Using 8 threads for script verification
2019-12-07T15:32:36Z scheduler thread start
2019-12-07T15:32:36Z HTTP: creating work queue of depth 16
2019-12-07T15:32:36Z No rpcpassword set - using random cookie authentication.
2019-12-07T15:32:36Z Generated RPC authentication cookie /home/login/Bitcoin/.cookie
2019-12-07T15:32:36Z HTTP: starting 4 worker threads
2019-12-07T15:32:36Z Using wallet directory /home/login/Bitcoin
... load wallets ...
2019-12-07T15:32:36Z init message: Loading banlist...
2019-12-07T15:32:36Z Cache configuration:
2019-12-07T15:32:36Z * Using 2.0 MiB for block index database
2019-12-07T15:32:36Z * Using 8.0 MiB for chain state database
2019-12-07T15:32:36Z * Using 440.0 MiB for in-memory UTXO set (plus up to 286.1 MiB of unused mempool space)
2019-12-07T15:32:36Z init message: Loading block index...
2019-12-07T15:32:36Z Opening LevelDB in /home/login/Bitcoin/blocks/index
2019-12-07T15:32:36Z Opened LevelDB successfully
2019-12-07T15:32:36Z Using obfuscation key for /home/login/Bitcoin/blocks/index: 0000000000000000
2019-12-07T15:32:39Z LoadBlockIndexDB: last block file = 1888
2019-12-07T15:32:39Z LoadBlockIndexDB: last block file info: CBlockFileInfo(blocks=61, size=60399390, heights=607007...607067, time=2019-12-07...2019-12-07)
2019-12-07T15:32:39Z Checking all blk files are present...
2019-12-07T15:32:39Z LoadBlockIndexDB(): Block files have previously been pruned
2019-12-07T15:32:39Z : You need to rebuild the database using -reindex to go back to unpruned mode.  This will redownload the entire blockchain.
Please restart with -reindex or -reindex-chainstate to recover.
2019-12-07T15:32:39Z Aborted block database rebuild. Exiting.
2019-12-07T15:32:39Z Shutdown: In progress...
2019-12-07T15:32:39Z scheduler thread interrupt
2019-12-07T15:32:39Z Shutdown: done

`bitcoin-qt` starts normally with the same configuration : 

2019-12-07T15:40:22Z Bitcoin Core version v0.19.0.1 (release build)
2019-12-07T15:40:22Z Assuming ancestors of block 00000000000000000005f8920febd3925f8272a6a71237563d78c2edfdd09ddf have valid signatures.
2019-12-07T15:40:22Z Setting nMinimumChainWork=000000000000000000000000000000000000000008ea3cf107ae0dec57f03fe8
2019-12-07T15:40:22Z Prune configured to target 190734 MiB on disk for block and undo files.
2019-12-07T15:40:22Z Using the 'sse4(1way),sse41(4way),avx2(8way)' SHA256 implementation
2019-12-07T15:40:22Z Using RdRand as an additional entropy source
2019-12-07T15:40:22Z Default data directory /home/login/.bitcoin
2019-12-07T15:40:22Z Using data directory /home/login/Bitcoin
2019-12-07T15:40:22Z Config file: /home/login/Bitcoin/bitcoin.conf
2019-12-07T15:40:22Z Using at most 125 automatic connections (1024 file descriptors available)
2019-12-07T15:40:22Z Using 16 MiB out of 32/2 requested for signature cache, able to store 524288 elements
2019-12-07T15:40:22Z Using 16 MiB out of 32/2 requested for script execution cache, able to store 524288 elements
2019-12-07T15:40:22Z Using 8 threads for script verification
2019-12-07T15:40:22Z scheduler thread start
2019-12-07T15:40:22Z HTTP: creating work queue of depth 16
2019-12-07T15:40:22Z No rpcpassword set - using random cookie authentication.
2019-12-07T15:40:22Z Generated RPC authentication cookie /home/login/Bitcoin/.cookie
2019-12-07T15:40:22Z HTTP: starting 4 worker threads
2019-12-07T15:40:22Z Using wallet directory /home/login/Bitcoin
2019-12-07T15:40:22Z init message: Vérification des porte-monnaie...
... load wallets...
2019-12-07T15:40:22Z init message: Chargement de la liste d’interdiction...
2019-12-07T15:40:22Z Cache configuration:
2019-12-07T15:40:22Z * Using 2.0 MiB for block index database
2019-12-07T15:40:22Z * Using 8.0 MiB for chain state database
2019-12-07T15:40:22Z * Using 3990.0 MiB for in-memory UTXO set (plus up to 286.1 MiB of unused mempool space)
2019-12-07T15:40:22Z init message: Chargement de l’index des blocs…
2019-12-07T15:40:22Z Opening LevelDB in /home/login/Bitcoin/blocks/index
2019-12-07T15:40:22Z Opened LevelDB successfully
2019-12-07T15:40:22Z Using obfuscation key for /home/login/Bitcoin/blocks/index: 0000000000000000
2019-12-07T15:40:25Z LoadBlockIndexDB: last block file = 1888
2019-12-07T15:40:25Z LoadBlockIndexDB: last block file info: CBlockFileInfo(blocks=61, size=60399390, heights=607007...607067, time=2019-12-07...2019-12-07)
2019-12-07T15:40:25Z Checking all blk files are present...
2019-12-07T15:40:25Z LoadBlockIndexDB(): Block files have previously been pruned
2019-12-07T15:40:25Z Opening LevelDB in /home/login/Bitcoin/chainstate
2019-12-07T15:40:25Z Opened LevelDB successfully
2019-12-07T15:40:25Z Using obfuscation key for /home/login/Bitcoin/chainstate: d5826331177f4c85
2019-12-07T15:40:25Z Loaded best chain: hashBestChain=00000000000000000008dc07079dc7649636a1a6e127f422a7b617edc15dfd90 height=607067 date=2019-12-07T15:17:46Z progress=0.999989
2019-12-07T15:40:25Z init message: Rebobinage des blocs...
2019-12-07T15:40:26Z init message: Vérification des blocs... 
2019-12-07T15:40:26Z Verifying last 6 blocks at level 3
2019-12-07T15:40:26Z [0%]...[16%]...[33%]...[50%]...[66%]...[83%]...[99%]...[DONE].
2019-12-07T15:40:28Z No coin database inconsistencies in last 6 blocks (17252 transactions)
2019-12-07T15:40:28Z  block index            5561ms
2019-12-07T15:40:28Z init message: Chargement du porte-monnaie…

The config file `bitcoin.conf` in my datadir `/home/login/Bitcoin` contains : 

server=1
rpcport=8332
disablewallet=0
wallet=Wallet
wallet=Wallet2

The config file in `/home/login/.bitcoin` contains only `datadir=/home/login/Bitcoin`
This shows that `bitcoin-qt` did not report that it has pruned the node, although it knows it.

**Comment by emilengler** [comment]
See #15936

**Comment by Pantamis** [comment]
Nice ! I didn't saw this PR, thank you for pointing it.
It seems that the hidden settings I talk about are somewhere in Qt files then.

**Comment by Pantamis** [comment]
More details on the configuration file's location on this website : https://dev-notes.eu/2018/04/amend-bitcoin-qt-persistent-settings-for-bitcoin-core/

TLDR : 

ocation of Config File: Linux
On Linux/Unix systems, the following files are used by default:

$HOME/.config/Bitcoin/Bitcoin-Qt.conf
$HOME/.config/Bitcoin-Qt.conf
For myself, running Bitcoin Core under Ubuntu 16.04 the config file is $HOME/.config/Bitcoin/Bitcoin-Qt.conf

Location of Config File: Mac OS
On Mac OS X check the following locations:

$HOME/Library/Preferences/com.Bitcoin
$HOME/Library/Preferences/com.Bitcoin
/Library/Preferences/com.Bitcoin
/Library/Preferences/com.Bitcoin
Location of Config File: Windows
On Windows, check the following registry paths:

HKEY_CURRENT_USER\Software\Bitcoin
HKEY_CURRENT_USER\Software\Bitcoin
HKEY_LOCAL_MACHINE\Software\Bitcoin
HKEY_LOCAL_MACHINE\Software\Bitcoin
